File: java/src/main/structural/composite/Leaf.java
package structural.composite;

// Leaf: Represents individual objects in the hierarchy
public class Leaf implements Component {
    private final String name;

    public Leaf(String name) {
        this.name = name;
    }

    @Override
    public void operation() {
        System.out.println("Leaf: " + name + " is performing an operation.");
    }
}

File: python/src/structural/composite/leaf.py
from .component import Component


class Leaf(Component):
    def operation(self) -> str:
        return "Leaf"

    def is_composite(self) -> bool:
        return False

# Example Usage:
# leaf = Leaf()
# print(leaf.operation())  # Output: Leaf
#
# Common Mistakes to Avoid:
# 1. Treating Leaf as a composite:
#    - Mistake: Calling add/remove on Leaf.
#    - Fix: Only call add/remove on composites that override those methods.
